### PR TITLE
Added backend project to the repository.

### PR DESCRIPTION
**Pull request by ddhanesha**
I have added a python/django backend project to the repository. Please take a look at the format and let me know if you have any questions or if you find any issues with it. I have updated the README.md file to add steps on how to start the backend server. This PR is not touching any frontend code.

### CODE REVIEW

File: backend/climateconnect_main/settings.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = env('SECRET_ENV')
+
+# SECURITY WARNING: don't run with debug turned on in production!
```
Comment (EvanHahn) [inline backend/climateconnect_main/settings.py]: Once this is merged, we should add an issue for this.

File: backend/climateconnect_main/settings.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/2.2/ref/settings/
+"""
+
+import os
+from dotenv import find_dotenv, load_dotenv
+load_dotenv(find_dotenv('.pyenv'))
```
Comment (EvanHahn) [inline backend/climateconnect_main/settings.py]: `.pyenv` feels like the wrong name for this file, to me. I think it'd be easy to get it confused with [the pyenv project](https://github.com/pyenv/pyenv). Why not use `.env`?

File: backend/climateconnect_main/settings.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = env('SECRET_ENV')
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = []
```
Comment (EvanHahn) [inline backend/climateconnect_main/settings.py]: Does this automatically include localhost?

File: README.md [hunk 4]
```diff
@@ -1,23 +1,43 @@
...
 
 ### To Deploy
 
 1. Run `yarn build` to build our the website
-2. ??? Rest of deploy procedure is work in progress
-
-## Backend
-
-Nothing here yet!
+2. ??? Rest of deploy procedure is work in progress. Backend deploy Comming soon!
```
Comment (EvanHahn) [inline README.md]: Nitpick: `Comming` → `coming`

File: README.md [hunk 5]
```diff
@@ -1,23 +1,43 @@
...
+2.  Run `pip install -r requirements.txt` to install all backend libararies.
+3.  Create `.pyenv` to include backend environment variables. [Connect with other colloablators to
+    get env variables]
+4.  Run `python manage.py makemigrations` to run django migrations. [This command is used for when
+    you first start or whenever you are adding or updating database models]
+5.  Run server using `python manage.py runserver`.
+
+### Frontend
+
+1. From main directory Run `yarn install` to download all NPM packages
```
Comment (EvanHahn) [inline README.md]: Nitpick: `NPM` → `npm`

File: README.md [hunk 6]
```diff
@@ -1,23 +1,43 @@
 # Climate Connect
 
-## Frontend
+## Pre-steps
 
-### Getting Started
+1.  Create postgres database with username and password. [You will be using this for backend
```
Comment (EvanHahn) [inline README.md]: We should remove `[]` as they have meaning in Markdown.

File: README.md [hunk 7]
```diff
@@ -1,23 +1,43 @@
...
+    ```
+3.  Clone github repository `git clone <URL>`
+
+### Backend
+
+Once you clone the repository start backend server.
+
+1.  Go to backend directory `cd backend`
+2.  Run `pip install -r requirements.txt` to install all backend libararies.
+3.  Create `.pyenv` to include backend environment variables. [Connect with other colloablators to
```
Comment (EvanHahn) [inline README.md]: Nitpick: `colloablators` → `collaborators`

File: backend/climateconnect_main/settings.py (hunk 1 above)
Comment (ddhanesha) [reply backend/climateconnect_main/settings.py]: Yes, I agree. cc/ @positiveimpact

File: backend/climateconnect_main/settings.py (hunk 2 above)
Comment (ddhanesha) [reply backend/climateconnect_main/settings.py]: We have `.env` file for our frontend project so I thought the file name should be different. I can rename it to `.backend_env` or something similar. What do you think?

File: backend/climateconnect_main/settings.py (hunk 3 above)
Comment (ddhanesha) [reply backend/climateconnect_main/settings.py]: Good point, Pushing a fix with env variable ALLOWED_HOSTS to include all possible hosts we want to configure.

**Comment by ddhanesha** [comment]
> Looks good other than my comments!
> 
> One quick question: how much did you change when comparing this to a new Django app? Most of it looked pretty standard but I want to make sure that I didn't miss anything that you customized (beyond installing the Rest framework).

Most of it is pretty standard. Here are the changes that I made. 

1) I created a django project called `climateconnect_main` and django app called `climateconnect_api`. To give more context django project includes settings, wsgi and urls. While django app we can create database models and APIs. The only thing I changed inside django app is I created directories for `views`, `models`, `tests` instead of files `views.py`, `models.py` and `tests.py`. I made this change for better management of all APIs, methods and models we write. 
2) I installed `psycopg2` for postgresql support.

File: backend/climateconnect_main/settings.py (hunk 2 above)
Comment (EvanHahn) [reply backend/climateconnect_main/settings.py]: I think `.backend_env` is clearer—let's do that.

Comment (ddhanesha) [reply backend/climateconnect_main/settings.py]: Done

**Comment by ddhanesha** [comment]
Merging this PR upon approval.

**Pull request merged**